src/main/java/hudson/plugins/git/UserMergeOptions.java

```diff
@@ -36,22 +37,25 @@
      * @param mergeStrategy merge strategy to be used
      */
     @Deprecated
-    public UserMergeOptions(String mergeRemote, String mergeTarget, String mergeStrategy) {
-        this(mergeRemote, mergeTarget, mergeStrategy, MergeCommand.GitPluginFastForwardMode.FF);
+    public UserMergeOptions(String mergeRemote, String mergeTarget, String mergeStrategy, Integer timeout) {
+        this(mergeRemote, mergeTarget, mergeStrategy, MergeCommand.GitPluginFastForwardMode.FF, timeout);
     }
 
     /**
      * @param mergeRemote remote name used for merge
      * @param mergeTarget remote branch to be merged into current branch
      * @param mergeStrategy merge strategy
      * @param fastForwardMode fast forward mode
+     * @param timeout timeout before merge aborts
+     *
      */
     public UserMergeOptions(String mergeRemote, String mergeTarget, String mergeStrategy,
-            MergeCommand.GitPluginFastForwardMode fastForwardMode) {
+            MergeCommand.GitPluginFastForwardMode fastForwardMode, Integer timeout) {
```

Comment:
Essentially the same here, If you want a new ctor you should to deprecate this ctor and make a new one with the new parameter. You shouldn't mess with what is public API as this will break users of this code.

Reply:
@res0nance ,yes I will keep this in mind.